src/main/scala/org/camunda/feel/impl/interpreter/FeelInterpreter.scala

```diff
@@ -538,37 +538,32 @@ class FeelInterpreter {
       implicit context: EvalContext): Val =
     atLeastOne(xs map (x => () => eval(x)), f)
 
-  private def atLeastOne(xs: List[() => Val], f: Boolean => Val): Val =
-    xs match {
-      case Nil => f(false)
-      case x :: xs =>
-        x() match {
-          case ValBoolean(true)  => f(true)
-          case ValBoolean(false) => atLeastOne(xs, f)
-          case other =>
-            atLeastOne(xs, f) match {
-              case ValBoolean(true) => f(true)
-              case _                => ValNull
-            }
+  private def atLeastOne(items: List[() => Val], f: Boolean => Val): Val = {
+    items.foldLeft(f(false)) {
+      case (ValBoolean(true), _) => f(true)
+      case (ValNull, item) =>
+        item() match {
+          case ValBoolean(true) => f(true)
+          case _                => ValNull
         }
+      case (_, item) => withBooleanOrNull(item(), f)
     }
+  }
```

Comment:
❓ I don't understand the method's signature. Can you explain why `f` is a function that takes a `Boolean` and returns a `Val`? I'd expected a `Predicate<Val>` (or similar) for f.

Reply:
The parameter `f` is the constructor for the return value `ValBoolean`. 

Currently, this construct is used in many places in the interpreter. I plan to simplify it but don't want to do it now :sweat_smile:

Reply:
I understand it now, thanks 👍